validator/sawtooth_validator/consensus/proxy.py

```diff
@@ -54,12 +117,14 @@ def register(self, engine_name, engine_version, connection_id):
 
     def activate_if_configured(self, engine_name, engine_version):
         # Wait until chain head is committed
-        chain_head = None
-        while chain_head is None:
-            try:
-                chain_head = self.chain_head_get()
-            except UnknownBlock:
-                pass
+        try:
+            chain_head = self.chain_head_get()
+        except UnknownBlock:
+            LOGGER.warning(
+                "Unable to activate engine %s %s until chain head received",
```

Comment:
This message seems to imply that the engine should be activated; "Cannot check engine activation until chain header received" or something to that effect would be more accurate.